assets/react/v3/entries/course-builder/components/layouts/Notebook.tsx

```diff
@@ -0,0 +1,366 @@
+		},
+	});
+
+	const { handleResize, isResizing, size } = useResize({
+		resizeDivRef: wrapperRef,
+		options: {
+			minHeight: notebook.MIN_NOTEBOOK_HEIGHT,
+			minWidth: notebook.MIN_NOTEBOOK_WIDTH,
+		},
+	});
+
+	const onContentBlur = (event: React.FocusEvent) => {
+		setContentEditable(false);
+		const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+		const tempDiv = document.createElement('div');
+		tempDiv.innerHTML = event.currentTarget.innerHTML;
+		const text = tempDiv.innerText || '';
+		setToLocalStorage(
+			LocalStorageKeys.notebook,
+			JSON.stringify({
+				...notebookData,
+				notes: text,
+			})
+		);
+	};
+
+	const handleMouseDown = (event: React.MouseEvent) => {
+		event.preventDefault();
+		event.stopPropagation();
+
+		if (isCollapsed) return;
+
+		setIsDragging(true);
+		const { clientX, clientY } = event;
+		const wrapper = wrapperRef.current;
+
+		if (wrapper) {
+			const rect = wrapper.getBoundingClientRect();
+			setOffset({ x: clientX - rect.left, y: clientY - rect.top });
+		}
+	};
+
+	useEffect(() => {
+		const handleMouseMove = throttle((event: MouseEvent) => {
+			if (!isDragging || !isDefined(wrapperRef.current)) return;
+			setIsFloating(true);
+
+			const warpper = wrapperRef.current;
+			const { offsetWidth: notebookWidth, offsetHeight: notebookHeight } = warpper;
+			const { innerWidth: windowWidth, innerHeight: windowHeight } = window;
+
+			const newX = Math.min(Math.max(event.clientX - offset.x, 0), windowWidth - notebookWidth);
+			const newY = Math.min(Math.max(event.clientY - offset.y, 0), windowHeight - notebookHeight);
+
+			warpper.style.left = `${newX}px`;
+			warpper.style.top = `${newY}px`;
+		}, 10);
+
+		const handleMouseUp = () => {
+			setIsDragging(false);
+
+			if (isDefined(wrapperRef.current)) {
+				const wrapper = wrapperRef.current;
+				const { left, top, height, width } = wrapper.style;
+				const currentNotebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+				if (left === 'auto' || top === 'auto') return;
+				setToLocalStorage(
+					LocalStorageKeys.notebook,
+					JSON.stringify({
+						...currentNotebookData,
+						position: {
+							left,
+							top,
+							height,
+							width,
+						},
+					})
+				);
+			}
+		};
+
+		document.addEventListener('mousemove', handleMouseMove);
+		document.addEventListener('mouseup', handleMouseUp);
+
+		return () => {
+			document.removeEventListener('mousemove', handleMouseMove);
+			document.removeEventListener('mouseup', handleMouseUp);
+		};
+	}, [isDragging, offset]);
+
+	// biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+	useEffect(() => {
+		if (isResizing && isDefined(wrapperRef.current)) {
+			const wrapper = wrapperRef.current;
+			const { left, top, height, width } = wrapper.style;
+			const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
```

Comment:
Write a utility function for parse JSON string.
```ts
const jsonParse = <T,>(data: string): T => {
    return JSON.parse(data) as T;
}

const value = jsonParse<{name: string}>('{"name": "hello"}');
```
Now you can pass the type as a generic value and the output value will be inferred from the generic.